pkg/controller/release/scheduler_rolloutblock.go

```diff
@@ -0,0 +1,52 @@
+package release
+
+import (
+	"fmt"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/util/runtime"
+
+	shipper "github.com/bookingcom/shipper/pkg/apis/shipper/v1alpha1"
+	"github.com/bookingcom/shipper/pkg/util/rolloutblock"
+)
+
+func (s *Scheduler) processRolloutBlocks(rel *shipper.Release) (bool, string) {
+	relOverrideRBs := rolloutblock.NewOverride(rel.Annotations[shipper.RolloutBlocksOverrideAnnotation])
+
+	nsRBs, err := s.rolloutBlockLister.RolloutBlocks(rel.Namespace).List(labels.Everything())
+	if err != nil {
+		runtime.HandleError(fmt.Errorf("failed to list rollout block objects: %s", err))
+	}
+
+	gbRBs, err := s.rolloutBlockLister.RolloutBlocks(shipper.GlobalRolloutBlockNamespace).List(labels.Everything())
+	if err != nil {
+		runtime.HandleError(fmt.Errorf("failed to list rollout block objects: %s", err))
+	}
+
+	existingRBs := rolloutblock.NewOverrideFromRolloutBlocks(append(nsRBs, gbRBs...))
+	obsoleteRbs := relOverrideRBs.Diff(existingRBs)
```

Comment:
This bit looks very similar to what happens in the application controller. Maybe we should DRY it up?

Reply:
Tried, it did not go well. It is *very* similar, but removing the duplications removed only 4 lines and introduce ambiguity...